### PR TITLE
Assign milestone only after merging a PR

### LINKED ISSUE #6516: Milestone should be assigned to PRs when they are merged, not when they are approved
In https://github.com/woocommerce/automations/pull/41 we added an automation to automatically assign the milestone to a PR once it's approved. However, it would be better to assign it once the PR is merged. The main reason is that PRs might be approved, but they might not be ready for release yet (ie: there are conflicts, the PR needs to be released with a pairing PR, the PR was pre-approved but there are still changes needed, etc.).

This issue is about updating the automation so PRs are assigned to the next milestone when and only when they are merged into `trunk`.

### PR DESCRIPTION
**Pull request by albarin**
The goal of this PR is to assign the milestone to a PR once it is merged. 

The main reason is that PRs might be approved, but they might not be ready for release yet (ie: there are conflicts, the PR needs to be released with a pairing PR, the PR was pre-approved but there are still changes needed, etc.).

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/6516

### Testing
I've tested with a test repo on my account https://github.com/albarin/test-action and the action was skipped when the PR was closed https://github.com/albarin/test-action/actions/runs/5750564158 and it run after the PR was merged https://github.com/albarin/test-action/actions/runs/5750595094. So the condition seems to be working, but until we merge I think we cannot be completely sure 🤔 

* [x] Do not include in the Testing Notes

### CODE REVIEW

**Review by thealexandrelara — APPROVED** [review]
I'm wondering about what would happen when the milestone was manually set by someone before merging the PR. For example, let's say I want to include the PR to a closed milestone, if I manually set it and then merge the PR, would it be replaced with the current milestone?

I'm pre-approving this since I think there is no easy way to test, but if this scenario was already tested or if these changes is already covering this, it's okay to merge it 🚀

**Comment by albarin** [comment]
> I'm wondering about what would happen when the milestone was manually set by someone before merging the PR. For example, let's say I want to include the PR to a closed milestone, if I manually set it and then merge the PR, would it be replaced with the current milestone?
> 
> I'm pre-approving this since I think there is no easy way to test, but if this scenario was already tested or if these changes is already covering this, it's okay to merge it 🚀

Thanks @thealexandrelara! I did not think about it, but checking at the `assign-milestone` automation I think it's already taken care of, it seems that it's skipped if the PR already has a milestone.
https://github.com/woocommerce/automations/blob/ccb50c64640a931cb0f4266398ce18216132b8a4/lib/automations/assign-milestone/pull-request-review-handler.js#L56-L62

**Comment by danielwrobert** [comment]
Can this be merged or do we still need to do some testing?

I see the PR was approved and has the `status: ready to merge` label but it's not clear why the `status: on hold` label was added.

**Comment by danielwrobert** [comment]
It looks like, based on this internal conversation (p1691681252116789/1691662190.610479-slack-C02UBB1EPEF), that we still need to do some more testing around how the current plugin version is identified.

I assume this is still pending that investigation?

**Pull request closed without merging**
